src/libfuncs/array.rs

```diff
@@ -1237,309 +1846,137 @@ pub fn build_slice<'ctx, 'this>(
     Ok(())
 }
 
-/// Generate MLIR operations for the `span_from_tuple` libfunc.
-pub fn build_span_from_tuple<'ctx, 'this>(
+/// Generate MLIR operations for the `array_len` libfunc.
+pub fn build_len<'ctx, 'this>(
     context: &'ctx Context,
-    registry: &ProgramRegistry<CoreType, CoreLibfunc>,
+    _registry: &ProgramRegistry<CoreType, CoreLibfunc>,
```

Comment:
Why we need this unused argument? Can we remove it?

Reply:
Yes, but I'd rather leave it for consistency. All libfunc implementations have the same arguments regardless of whether they're being used or not.

It's convenient when we have to fix something and all the data we need is already available, even if we weren't using it before.